v3/src/components/graph/components/dot-chart-bars.tsx

```diff
@@ -0,0 +1,234 @@
+import { ScaleBand, ScaleLinear, select } from "d3"
+import { observer } from "mobx-react-lite"
+import React, { useCallback, useEffect, useRef } from "react"
+import { createPortal } from "react-dom"
+import { CellType, IBarCover, PlotProps } from "../graphing-types"
+import { usePlotResponders } from "../hooks/use-plot"
+import { handleClickOnBar } from "../../data-display/data-display-utils"
+import { setPointCoordinates } from "../utilities/graph-utils"
+import { IDataConfigurationModel } from "../../data-display/models/data-configuration-model"
+import { GraphLayout } from "../models/graph-layout"
+import { SubPlotCells } from "../models/sub-plot-cells"
+import { mstAutorun } from "../../../utilities/mst-autorun"
+import { useChartDots } from "../hooks/use-chart-dots"
+
+interface IRenderBarCoverProps {
+  barCovers: IBarCover[]
+  barCoversRef: React.RefObject<SVGGElement>
+  dataConfig: IDataConfigurationModel
+  primaryAttrRole: "x" | "y"
+}
+
+interface IBarCoverDimensionsProps {
+  subPlotCells: SubPlotCells
+  cellIndices: CellType
+  layout: GraphLayout
+  maxInCell: number
+  minInCell?: number
+  primCatsCount: number
+}
+
+const renderBarCovers = (props: IRenderBarCoverProps) => {
+  const { barCovers, barCoversRef, dataConfig, primaryAttrRole } = props
+  select(barCoversRef.current).selectAll("rect").remove()
+  select(barCoversRef.current).selectAll("rect")
+    .data(barCovers)
+    .join((enter) => enter.append("rect")
+      .attr("class", (d) => d.class)
+      .attr("data-testid", "bar-cover")
+      .attr("x", (d) => d.x)
+      .attr("y", (d) => d.y)
+      .attr("width", (d) => d.width)
+      .attr("height", (d) => d.height)
+      .on("mouseover", function() { select(this).classed("active", true) })
+      .on("mouseout", function() { select(this).classed("active", false) })
+      .on("click", function(event, d) {
+        dataConfig && handleClickOnBar({event, dataConfig, primaryAttrRole, barCover: d})
+      })
+    )
+}
+
+const barCoverDimensions = (props: IBarCoverDimensionsProps) => {
+  const { subPlotCells, cellIndices, layout, maxInCell, minInCell = 0, primCatsCount } = props
+  const {
+    primaryAxisPlace, primaryIsBottom, primarySplitAxisPlace, primarySplitCellWidth,
+    secondaryAxisPlace, secondaryCellHeight, numSecondarySplitBands
+  } = subPlotCells
+  const { p: primeCatIndex, ep: primeSplitCatIndex, es: secSplitCatIndex } = cellIndices
+  const primarySplitAxisScale = layout.getAxisScale(primarySplitAxisPlace) as ScaleBand<string>
+  const secondaryAxisScale = layout.getAxisScale(secondaryAxisPlace) as ScaleLinear<number, number>
+  const numPrimarySplitBands = Math.max(1, primarySplitAxisScale?.domain().length ?? 1)
+  const primaryCellWidth = layout.getAxisLength(primaryAxisPlace) / (primCatsCount ?? 1)
+  const primarySubCellWidth = primaryCellWidth / numPrimarySplitBands
```

Comment:
Some of these are either already available in the `SubPlotCells` or could be made available there. Again, not required for this PR but something to consider moving forward.

Reply:
Thanks for the reminder about that. I've cleaned this part up a bit as suggested.